tests/test_combine_design_matrices_by_param.py

```diff
@@ -0,0 +1,28 @@
+"""Test combine_design_matrices_by_param function"""
+
+import astropy.units as u
+import pytest
+from pint.pint_matrix import DesignMatrixMaker, combine_design_matrices_by_param
+from pint.config import examplefile
+from pint.models import get_model_and_toas
+
+
+@pytest.fixture()
+def model_and_toas():
+    parfile = examplefile("B1855+09_NANOGrav_dfg+12_TAI.par")
+    timfile = examplefile("B1855+09_NANOGrav_dfg+12.tim")
+    model, toas = get_model_and_toas(parfile, timfile)
+    return model, toas
+
+
+def test_combine_design_matrices_by_param(model_and_toas):
```

Comment:
What property of the function does this test verify? If it's verifying more than one, please split it into separate tests so we can tell where failures occur.

Reply:
I have renamed this test and added a docstring to explain what is being tested.